plugins/BEdita/API/tests/IntegrationTest/NewObjectTypesTest.php

```diff
@@ -0,0 +1,142 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2017 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+namespace BEdita\API\Test\IntegrationTest;
+
+use BEdita\API\TestSuite\IntegrationTestCase;
+use Cake\ORM\TableRegistry;
+
+/**
+ * Test new object types creation, along with objects implementations
+ */
+class NewObjectTypesTest extends IntegrationTestCase
+{
+    /**
+     * {@inheritDoc}
+     */
+    public $fixtures = [
+        'plugin.BEdita/Core.media',
+        'plugin.BEdita/Core.streams',
+    ];
+
+    /**
+     * Data provider for `testNewTypes`
+     */
+    public function newTypesProvider()
+    {
+        return [
+            'cats' => [
+                [
+                    'name' => 'cats',
+                    'singular' => 'cat',
+                    'plugin' => 'BEdita/Core',
+                    'model' => 'Objects',
+                ],
+                [
+                    'description' => 'another cat',
+                ]
+            ],
+            'songs' => [
+                [
+                    'name' => 'songs',
+                    'singular' => 'song',
+                    'plugin' => 'BEdita/Core',
+                    'model' => 'Media',
+                    'parent_name' => 'media',
+                ],
+                [
+                    'description' => 'a new song',
+                ]
+            ],
+        ];
+    }
+
+    /**
+     * Test new object type creation with object implementation
+     *
+     * @param $typeData array New object type intpu data
+     * @param $attributes array New object attributes
+     * @dataProvider newTypesProvider
+     * @coversNothing
+     */
+    public function testNewTypes($typeData, $attributes)
+    {
+        $lastId = $this->lastObjectId();
```

Comment:
This test case covers many operations. If one fails the whole test fails, making it difficult to spot the real cause of the error at first sight, and possibly hiding other failures. I'd prefer to split this test case in multiple test cases.

Reply:
That's true, but since it's an integration test (with zero coverage) not a unit test I don't see particular problems here. 
Testing this sequence of API calls is the actual goal.